sql/distsql: TestAggregatorAgainstProcessor failed

sql/distsql.TestAggregatorAgainstProcessor [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4159957&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4159957&tab=artifacts#/) on release-21.1 @ [4d2b93943c4333cfeabbe240465926f3b1f9e724](https://github.com/cockroachdb/cockroach/commits/4d2b93943c4333cfeabbe240465926f3b1f9e724):


```
(43, NULL, 43, ('1984-08-09', (), NULL, '*'), 1790796742, 1.093726262002816, -4697831285357933053, +Inf, 1662304848, NULL),
(9, 9, 10, ('1992-03-11', (), 4205859320, NULL), 1241892955, 0.711474899516769, -7177368186075542776, +Inf, 34354077, 4704),
(19, 19, 19, ('1972-12-03', (), 136823566, '"'), -323972846, -0.4447344909379799, -7831540207638910313, -1.100239634513855, 3753731856, NULL),
(32, 32, 32, ('1975-04-17', (), NULL, 'y'), -72339796, 5e-324, -1825304802216470468, -0.32055380940437317, 4237077154, -28716),
(15, 16, 16, ('1972-01-11', (), 2782473446, 'W'), -1582060168, NULL, 1922442150236030404, -1.7239460945129395, 4120415683, -741),
(39, 40, 41, ('1976-10-08', (), 4170698674, 'X'), 988943735, 3.4028234663852886e+38, 4295786174287297657, -1.6456022262573242, 1916767935, -14461),
(20, 20, 20, ('1977-09-12', (), NULL, 'l'), -982236525, -0.9284021164468877, -9223372036854775807, 0.47356492280960083, 725436063, 7976),
(27, 27, 27, ('1997-01-06', NULL, 3594093585, e'\x12'), 1135458612, 0.33994978189685854, 2147483647, 1.4221776723861694, 1314464570, 11872),
(32, NULL, 32, ('1996-06-24', (), 0, 'Y'), -1614756270, -0.18798997963959468, NULL, 0.5340198278427124, 3932802672, -17226),
(52, 52, 52, ('1977-11-16', (), 0, NULL), 2064099256, 0.02641767483962998, 991278099232829007, 1.1852319240570068, 3272147013, NULL),
(45, 45, NULL, ('1987-01-22', (), 567534515, '('), -109356534, 1.9984158207737983, 1776828034037215447, +Inf, 2296832800, -11415),
(NULL, 18, 18, ('1978-12-27', (), 230995155, 'u'), 2037836767, -0.4094796349961288, -2147483648, -1.6797144412994385, 2319356227, -21030),
(24, 24, 24, ('1994-12-04', (), 803354206, ''), 347839252, -0.44019222719208884, -6513344942016502786, 1.0966403484344482, 1875960587, 12845),
(18, NULL, 18, ('1983-07-02', (), 3608313012, NULL), -1, NULL, 1927656299636552960, 3.4028234663852886e+38, 1597161977, -963),
(24, 24, 24, ('1995-11-29', (), 0, ' '), -182228904, -1.7321088555209752, 5709212655503154215, -0.8024347424507141, 903921174, 7674),
(49, NULL, 50, ('1971-05-13', (), 1943410143, '&'), NULL, 1.401298464324817e-45, -3902612272669756225, 0.09976794570684433, 0, -11941),
(26, 26, 26, (NULL, (), 0, e'\x1c'), 0, -0.28954320210615087, NULL, +Inf, 747481976, -30776),
(33, 33, 33, ('1984-05-01', (), 3152955892, 'H'), NULL, -0.12433109453023672, NULL, NULL, 1419610102, -10327),
(37, 38, 39, NULL, -968204404, -0.9223369343859495, -32768, -0.7870498299598694, 676299914, -3681);
    columnar_operators_test.go:327: different number of metas returned:
        processor returned
        [{Ranges:[] Err:cannot append tuple{date, tuple, regtype, "char"} to array containing tuple{date, unknown, oid, string}
        (1) assertion failure
        Wraps: (2) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*DArray).Append
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:3975
          | github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.(*arrayAggregate).Add
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/aggregate_builtins.go:1319
          | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*aggregatorBase).accumulateRowIntoBucket
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:796
          | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRow
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:859
          | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).accumulateRows
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:441
          | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*hashAggregator).Next
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/aggregator.go:692
          | github.com/cockroachdb/cockroach/pkg/sql/distsql.verifyColOperator
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_utils_test.go:202
          | github.com/cockroachdb/cockroach/pkg/sql/distsql.TestAggregatorAgainstProcessor
          | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/columnar_operators_test.go:308
          | testing.tRunner
          | 	/usr/local/go/src/testing/testing.go:1123
          | runtime.goexit
          | 	/usr/local/go/src/runtime/asm_amd64.s:1374
        Wraps: (3) cannot append tuple{date, tuple, regtype, "char"} to array containing tuple{date, unknown, oid, string}
        Error types: (1) *assert.withAssertionFailure (2) *withstack.withStack (3) *errutil.leafError TraceData:[] LeafTxnFinalState:<nil> RowNum:<nil> SamplerProgress:<nil> BulkProcessorProgress:<nil> Metrics:<nil>}]
        columnar operator returned
        []
--- FAIL: TestAggregatorAgainstProcessor (2.24s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestAggregatorAgainstProcessor PKG=./pkg/sql/distsql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/sql-queries @yuzefovich
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggregatorAgainstProcessor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/distsql/columnar_operators_test.go
+++ b/pkg/sql/distsql/columnar_operators_test.go
@@ -224,7 +224,8 @@ func TestAggregatorAgainstProcessor(t *testing.T) {
 									execinfrapb.StMakeline,
 									execinfrapb.StExtent,
 									execinfrapb.StUnion,
-									execinfrapb.StCollect:
+									execinfrapb.StCollect,
+									execinfrapb.ArrayAgg:
 									for _, typ := range aggFnInputTypes {
 										if typ.Family() == types.TupleFamily || (typ.Family() == types.ArrayFamily && typ.ArrayContents().Family() == types.TupleFamily) {
 											invalid = true

